From 47eca84cc31fc905cb91c7652bdeb11e52d0ceac Mon Sep 17 00:00:00 2001
From: Diogo Ferreira <diogo@underdev.org>
Date: Fri, 14 Sep 2018 10:51:15 +0200
Subject: [PATCH 18/32] proguard: Also keep FixedScaleDrawable

FixedScaleDrawable is used for legacy icons if LEGACY_ICON_TREATMENT is
set to false. However, it gets removed by proguard, this patch adds a
keep rule.

Change-Id: Ifcedad251013abefbb52cd7bc0ed2f59aa1bc06f
Signed-off-by: Alexander Martinz <amartinz@shiftphones.com>
---
 proguard.flags | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/proguard.flags b/proguard.flags
index e4011165e..93b0d34d8 100644
--- a/proguard.flags
+++ b/proguard.flags
@@ -76,6 +76,10 @@
   public void setBackgroundAlpha(float);
 }
 
+-keep class com.android.launcher3.graphics.FixedScaleDrawable {
+  public <init>(...);
+}
+
 # Proguard will strip new callbacks in LauncherApps.Callback from
 # WrappedCallback if compiled against an older SDK. Don't let this happen.
 -keep class com.android.launcher3.compat.** {
-- 
2.17.1

